SystemMicrocontroller_Sensors/Core/Src/main.c

```diff
@@ -33,251 +34,299 @@
 
 
 /* Private variables ---------------------------------------------------------*/
-RTC_HandleTypeDef hrtc;
+LCD_HandleTypeDef hlcd;
 
 /* Private function prototypes -----------------------------------------------*/
 void SystemClock_Config(void);
 static void MX_GPIO_Init(void);
+void MX_LCD_Init(void);
 
 
 /* Private user code ---------------------------------------------------------*/
 
 
 /**
-  * @brief  The application entry point.
-  * @retval int
-  */
+ * @brief  The application entry point.
+ * @retval int
+ */
 int main(void)
 {
-  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
-  HAL_Init();
+	/* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+	HAL_Init();
 
-  /* Configure the system clock */
-  SystemClock_Config();
+	/* Configure the system clock */
+	SystemClock_Config();
 
-  /* Initialize all configured peripherals */
-  MX_GPIO_Init();
+	/* Initialize all configured peripherals */
+	//MX_GPIO_Init();
 
+	MX_LCD_Init();
 
-  while (1)
-  {
+	LCD_Clear(&hlcd);
+	LCD_PrintTo1stLine(&hlcd, "Testing 20x4 LCD");
+	LCD_SetCursorPosition(&hlcd, 1, 4);
+	LCD_Printf(&hlcd, "Hello World!");
+	LCD_PrintTo4thLine(&hlcd, "Today's date: May %d", 16);
```

Comment:
A question here, does the processor get blocked while writing to the LCD? Should we consider adding a operating system? Or can we do this interrupt based?

Reply:
Processor does get blocked while writing to LCD. Definitely something to consider in the application side.

If needed, we could always code an interrupt-based implementation using a data buffer and a timer